app/Http/Controllers/TradeController.php

```diff
@@ -15,7 +16,28 @@ class TradeController extends Controller
      */
     public function index(Request $request)
     {
-        $trades = Trade::all();
+        $trades = Trade::where('status', 'open')
+                    ->has('items')->with('items')
+                    ->filtered($request->get('requested_items'))
+                    ->filtered($request->get('description'));
+
+        if ($itemName = $request->get('item_name')) {
+            $trades = $trades->whereHas('items', function (Builder $q) use ($itemName) {
+                $q->where('name', 'like', "%{$itemName}%");
+            });
+        }
+
+        if ($itemDescription = $request->get('item_description')) {
+            $trades = $trades->whereHas('items', function (Builder $q) use ($itemDescription) {
+                $q->where('description', 'like', "%{$itemDescription}%");
+            });
+        }
+
+        if ($itemRarity = $request->get('item_rarity')) {
+            $trades = $trades->whereHas('items', function (Builder $q) use ($itemRarity) {
```

Comment:
these `whereHas` should be on the `item` relation, since thats what is being offered 
```suggestion
            $trades = $trades->whereHas('item', function (Builder $q) use ($itemName) {
                $q->where('name', 'like', "%{$itemName}%");
            });
        }

        if ($itemDescription = $request->get('item_description')) {
            $trades = $trades->whereHas('item', function (Builder $q) use ($itemDescription) {
                $q->where('description', 'like', "%{$itemDescription}%");
            });
        }

        if ($itemRarity = $request->get('item_rarity')) {
            $trades = $trades->whereHas('item', function (Builder $q) use ($itemRarity) {
```